### PR TITLE
Update to file-leak-detector:1.11 to detect pipe and selector leaks

### PR DESCRIPTION
**Pull request by dwnusbaum**
Subsumes #2. 

Updates the parent pom and specifies explicit Jenkins and Java versions, pulls in https://github.com/kohsuke/file-leak-detector/pull/34 which adds support for detecting `Pipe`s, `Selector`s, and files opened via `FileChannel`,  and adds some basic tests.

@reviewbybees

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 target
 work
+.idea
```
Comment (denisj44) [inline .gitignore]: I don't think idea editor specific files needs to be added.

Comment (dwnusbaum) [reply .gitignore]: I think you added it originally in https://github.com/jenkinsci/file-leak-detector-plugin/pull/3/commits/504dc08556e22aff4ea8083aecd422bbde2eab67 😄

I will go ahead and remove it, but for what it's worth `.idea` and other IntelliJ/Eclipse files are added to `.gitignore` by default for any plugins using the maven archetype generator: https://github.com/jenkinsci/archetypes/blob/master/empty-plugin/src/main/resources/archetype-resources/.gitignore. We could follow up with a separate PR to add all of those entries to `.gitignore` if desired.

**Comment by dwnusbaum** [comment]
@reviewbybees done

File: src/test/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDumpTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    /**
+     * Activates the file-leak-detector library. Only activates once for the
+     * whole test suite, but is called in every test so they can run in any
+     * order, and cannot not be called until Jenkins starts.
+     */
+    private void activateFileLeakDetector() throws Exception {
+        FileHandleDump fileHandleDump = ManagementLink.all().get(FileHandleDump.class);
+        fileHandleDump.doActivate("");
```
Comment (jglick) [inline src/test/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDumpTest.java]: [Fails](https://ci.jenkins.io/job/Plugins/job/file-leak-detector-plugin/job/PR-4/1/testReport/com.cloudbees.jenkins.plugins.file_leak_detector/FileHandleDumpTest/detectFileLeak/) on Windows.

Comment (dwnusbaum) [reply src/test/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDumpTest.java]: Good catch. I'll investigate to see whether we can fix it here or if the library itself doesn't work on Windows.

Comment (jglick) [reply src/test/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDumpTest.java]: I really have no idea. You can always

```groovy
buildPlugin(platforms: ['linux'])
```

if it is _known_ not to work on Windows.

Comment (dwnusbaum) [reply src/test/java/com/cloudbees/jenkins/plugins/file_leak_detector/FileHandleDumpTest.java]: Updating to JNR fixed the main issue, but pipes aren't detected correctly on Windows by the library, so I've ignored that test on Windows for now. See https://github.com/kohsuke/file-leak-detector/issues/36.

**Pull request merged**